docs/reference/files/file-charmcraft-yaml.rst

```diff
@@ -0,0 +1,901 @@
+.. _file-charmcraft-yaml:
+.. highlight:: yaml
+
+``charmcraft.yaml``
+*******************
+
+..
+    AUTHOR NOTE:
+    The full list of keys is defined in the Charmcraft project (but this implies upstream keys from craft-application):
+    https://github.com/canonical/charmcraft/blob/3.2.0/charmcraft/models/project.py#L381-L1070
+
+    Technically, the only key required upfront is type. But then, depending on what you choose, other keys become required as well. (The required keys are the ones that are set to a value, unless that value is a pydantic.Field that doesn't have either a default or a default_factory parameter)
+
+    The parts key connects to an external library. The plugin properties are defined here: https://canonical-craft-parts.readthedocs-hosted.com/en/latest/reference/part_properties.html
+
+    This test file shows the full spec at once: https://github.com/canonical/charmcraft/blob/main/tests/unit/models/valid_charms_yaml/full.yaml
+
+.. important::
+    ``charmcraft.yaml`` file is the only ``yaml`` file generated by ``charmcraft init`` and the only ``yaml`` file in a charm project that a charm author should edit directly.
+
+    Charmcraft will use the information you provide here to generate :ref:`file-actions-yaml`, :ref:`file-config-yaml`, and :ref:`file-metadata-yaml`, as well as all the other files it usually does.
+
+``charmcraft.yaml`` is a file in your charm project that contains keys that allow you to declare information about the project in a form that can be used by Charmcraft.
+
+.. note::
+    If you're starting from an empty file, the only required key is the ``type`` key. However, depending on what value you set it to (``charm`` or ``bundle``), other keys become required as well.
+
+
+.. dropdown:: Expand to view a full charm with sample content all at once
```

Comment:
In other Starcraft and Store projects we use [`collapse`](https://github.com/canonical/snapcraft/blob/a34296875e9b90d7c27632d49407993bca26e649/docs/conf.py#L55) because it's [less visually jarring](https://canonical-snapcraft.readthedocs-hosted.com/en/latest/howto/craft-a-snap/example-pre-built-app/#example-recipe-for-geekbench).

Reply:
```suggestion
.. collapse:: Expand to view a full charm with sample content all at once
```